### PR TITLE
Fix border on android buttons 

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change ###

Don't allow negative border width .
Refactoring and add support for border properties to appcompact and fast renderer button  

### Bugs Fixed ###

https://bugzilla.xamarin.com/show_bug.cgi?id=36031

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/AppCompat/ButtonRenderer.cs [hunk 1]
```diff
@@ -27,6 +23,7 @@ public class ButtonRenderer : ViewRenderer<Button, AppCompatButton>, AView.IOnAt
 
 		public ButtonRenderer()
 		{
+			System.Diagnostics.Debug.WriteLine("Slow Button!");
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.Android/AppCompat/ButtonRenderer.cs]: ?

**Comment by StephaneDelcroix** [comment]
you comment says 

> Don't allow negative border width

but the code only checks for negative `cornerRadius`

File: Xamarin.Forms.Platform.Android/AppCompat/ButtonRenderer.cs (hunk 1 above)
Comment (rmarinho) [inline Xamarin.Forms.Platform.Android/AppCompat/ButtonRenderer.cs]: This is something we also added on non app compact button so users select fast renderers, should we remove it @ez ?

**Comment by rmarinho** [comment]
@StephaneDelcroix my comment referes to border width (inset) calculation inside the renderer where if border radius was 0 it will have a negative value for the border width (inset) when drawing

File: Xamarin.Forms.Platform.Android/AppCompat/ButtonRenderer.cs (hunk 1 above)
Comment (hartez) [inline Xamarin.Forms.Platform.Android/AppCompat/ButtonRenderer.cs]: Yeah, that should have been removed a while ago.

**Pull request merged**